api/api.go

```diff
@@ -765,48 +760,77 @@ func (api *Server) GetActionByActionHash(h hash.Hash256) (action.SealedEnvelope,
 func (api *Server) GetEvmTransfersByActionHash(ctx context.Context, in *iotexapi.GetEvmTransfersByActionHashRequest) (*iotexapi.GetEvmTransfersByActionHashResponse, error) {
-	if !api.hasActionIndex || api.systemLogIndexer == nil {
-		return nil, status.Error(codes.Unavailable, "evm transfer index not supported")
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByActionHash instead")
+}
+
+// GetEvmTransfersByBlockHeight returns evm transfers by block height
+func (api *Server) GetEvmTransfersByBlockHeight(ctx context.Context, in *iotexapi.GetEvmTransfersByBlockHeightRequest) (*iotexapi.GetEvmTransfersByBlockHeightResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "evm transfer index is deprecated, call GetSystemLogByBlockHeight instead")
+}
+
+// GetSystemLogByActionHash returns system log by action hash
+func (api *Server) GetSystemLogByActionHash(
+	ctx context.Context,
+	in *iotexapi.GetSystemLogByActionHashRequest) (*iotexapi.GetSystemLogByActionHashResponse, error) {
+	if !api.hasActionIndex || api.indexer == nil {
+		return nil, status.Error(codes.Unimplemented, blockindex.ErrActionIndexNA.Error())
+	}
+	if !api.dao.ContainsSystemLog() {
+		return nil, status.Error(codes.Unimplemented, blockdao.ErrNotSupported.Error())
 	}
 
-	actHash, err := hash.HexStringToHash256(in.ActionHash)
+	h, err := hex.DecodeString(in.ActionHash)
 	if err != nil {
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	transfers, err := api.systemLogIndexer.GetEvmTransfersByActionHash(actHash)
+	actIndex, err := api.indexer.GetActionIndex(h)
+	if err != nil {
+		return nil, status.Error(codes.Internal, err.Error())
+	}
+
+	sysLog, err := api.dao.GetSystemLog(actIndex.BlockHeight())
 	if err != nil {
 		if errors.Cause(err) == db.ErrNotExist {
-			return nil, status.Error(codes.NotFound, "no such action with evm transfer")
+			return nil, status.Error(codes.NotFound, err.Error())
 		}
 		return nil, status.Error(codes.Internal, err.Error())
 	}
 
-	return &iotexapi.GetEvmTransfersByActionHashResponse{ActionEvmTransfers: transfers}, nil
+	for _, log := range sysLog.ActionSystemLog {
+		if in.ActionHash == hex.EncodeToString(log.ActionHash) {
+			return &iotexapi.GetSystemLogByActionHashResponse{
+				ActionSystemLog: log,
+			}, nil
+		}
+	}
+	return nil, status.Errorf(codes.NotFound, "system log not found for action %s", in.ActionHash)
 }
 
-// GetEvmTransfersByBlockHeight returns evm transfers by block height
-func (api *Server) GetEvmTransfersByBlockHeight(ctx context.Context, in *iotexapi.GetEvmTransfersByBlockHeightRequest) (*iotexapi.GetEvmTransfersByBlockHeightResponse, error) {
-	if !api.hasActionIndex || api.systemLogIndexer == nil {
-		return nil, status.Error(codes.Unavailable, "evm transfer index not supported")
+// GetSystemLogByBlockHeight returns system log by block height
+func (api *Server) GetSystemLogByBlockHeight(
```

Comment:
for external, is "SystemLog" a confusing name?

Reply:
fixed